tests/graph/test_graph.py

```diff
@@ -61,124 +54,166 @@ def create_simple_edge_list_from_graph(
-        input=input_edgelist,
-        output=output_dir,
-        output_prefix="test",
-        metrics_file=metrics_file,
-        multiplet_recovery=False,
-        min_count=1,
+@pytest.mark.parametrize("enable_backend", ["igraph", "networkx"], indirect=True)
+def test_build_graph_full_bipartite_do_not_simplify(
+    enable_backend,
+    full_graph_edgelist: pd.DataFrame,
+):
+    edgelist_with_multiedges = full_graph_edgelist.copy()
+    # Duplicate one row to create a multiedge
+    one_row = edgelist_with_multiedges.iloc[0].to_frame().T
+    one_row["umi"] = random_sequence(6)
+    edgelist_with_multiedges = pd.concat(
+        [one_row, edgelist_with_multiedges],
+        axis=0,
+        ignore_index=True,
     )
 
-    result_pixel_data_file = next(output_dir.glob("*.edgelist.csv.gz"))
-    result = pd.read_csv(result_pixel_data_file)
-    assert len(result["component"].unique()) == 1
-
-
-def test_recovery_technical_multiplets(
-    edgelist_with_communities: pd.DataFrame,
-):
-    """Test recovery of technical multiplet components."""
-    assert len(edgelist_with_communities["component"].unique()) == 1
+    # When not simplifying all edges should be kept
+    graph = Graph.from_edgelist(
+        edgelist=edgelist_with_multiedges,
+        add_marker_counts=False,
+        simplify=False,
+        use_full_bipartite=True,
+    )
+    assert graph.vcount() == 100
+    assert graph.ecount() == 2501
+    assert graph.vs.attributes() == {"name", "type", "pixel_type"}
 
-    result, info = recover_technical_multiplets(
-        edgelist=edgelist_with_communities.copy(),
+    # And the duplicate edge should disappear when we simplify
+    graph = Graph.from_edgelist(
+        edgelist=edgelist_with_multiedges,
+        add_marker_counts=False,
+        simplify=True,
+        use_full_bipartite=True,
     )
-    assert len(result["component"].unique()) == 2
-    assert info.keys() == {"PXLCMP0000000"}
-    assert sorted(list(info.values())[0]) == ["RCVCMP0000000", "RCVCMP0000001"]
+    assert graph.vcount() == 100
+    assert graph.ecount() == 2500
+    assert graph.vs.attributes() == {"name", "type", "pixel_type"}
 
 
-def test_community_detection_crossing_edges(graph_with_communities):
-    """Test discovery of crossing edges from graph with communities."""
-    result = community_detection_crossing_edges(
-        graph=graph_with_communities,
-        leiden_iterations=2,
+@pytest.mark.parametrize("enable_backend", ["igraph", "networkx"], indirect=True)
+def test_build_graph_a_node_projected(
+    enable_backend, full_graph_edgelist: pd.DataFrame
+):
+    """Build an A-node projected graph."""
+    graph = Graph.from_edgelist(
+        edgelist=full_graph_edgelist,
+        add_marker_counts=True,
+        simplify=True,
+        use_full_bipartite=False,
     )
-    assert result == [{"CTCGTACCTGGGACTGATACT", "TGTAAGTCAGTTGCAGGTTGG"}]
+    assert graph.vcount() == 50
+    assert graph.ecount() == ((50 * 50) / 2) - (50 / 2)
```

Comment:
Every time I look at this test I wonder how do we enforce this. And I have to look at the definition of `full_graph_edgelist` that is a fully connected bipartite graph. I wonder if we could put that into something. A comment before this assert or something like that.